src/primitives/transaction.h

```diff
@@ -78,23 +91,54 @@ static inline size_t JOINSPLIT_SIZE(int transactionVersion) {
     return transactionVersion >= SAPLING_TX_VERSION ? 1698 : 1802;
 }
 
+/**
+ * The storage format for Sapling Spend descriptions in v5 transactions.
+ */
+class SpendDescriptionV5
+{
+public:
+    uint256 cv;                    //!< A value commitment to the value of the input note.
+    uint256 nullifier;             //!< The nullifier of the input note.
+    uint256 rk;                    //!< The randomized public key for spendAuthSig.
+
+    SpendDescriptionV5() { }
+
+    SpendDescriptionV5(uint256 cv, uint256 nullifier, uint256 rk)
+        : cv(cv), nullifier(nullifier), rk(rk) { }
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITE(cv);
+        READWRITE(nullifier);
+        READWRITE(rk);
+    }
+};
+
 /**
  * A shielded input to a transaction. It contains data that describes a Spend transfer.
  */
-class SpendDescription
+class SpendDescription : public SpendDescriptionV5
 {
 public:
     typedef std::array<unsigned char, 64> spend_auth_sig_t;
 
-    uint256 cv;                    //!< A value commitment to the value of the input note.
     uint256 anchor;                //!< A Merkle root of the Sapling note commitment tree at some block height in the past.
-    uint256 nullifier;             //!< The nullifier of the input note.
-    uint256 rk;                    //!< The randomized public key for spendAuthSig.
     libzcash::GrothProof zkproof;  //!< A zero-knowledge proof using the spend circuit.
     spend_auth_sig_t spendAuthSig; //!< A signature authorizing this spend.
 
     SpendDescription() { }
 
+    SpendDescription(
+        uint256 cv,
+        uint256 anchor,
+        uint256 nullifier,
+        uint256 rk,
+        libzcash::GrothProof zkproof,
+        spend_auth_sig_t spendAuthSig)
+        : SpendDescriptionV5(cv, nullifier, rk), anchor(anchor), zkproof(zkproof), spendAuthSig(spendAuthSig) { }
+
     ADD_SERIALIZE_METHODS;
```

Comment:
Is this going to get the serialization of the anchor correct for V4 transactions?

Reply:
It will do once I'm done 😁

Reply:
Looked at this comment again at a more reasonable hour, and AFAICT it should already be doing so. `SerializationOp` in `SpendDescription` overrides the one in `SpendDescriptionV5`, and the v4 parsing logic is unaltered. For v5, the anchor is parsed into `SaplingBundle` and then explicitly copied into each `SpendDescription` in-memory.